Tombstone GC bin sharing is broken by monotonic time

Even after fiddling timing and reordering things to try to be more reliable in https://github.com/hashicorp/consul/commit/7b966e2d26833c64ec54a06ebdc027b2689a32ec, this test is failing unexpectedly:

```
[02:12:57][Step 1/1] --- FAIL: TestTombstoneGC (0.05s)
[02:12:57][Step 1/1] 	tombstone_gc_test.go:74: bad index: 120
```

Doing some debug tracing it looks like the time calculations are right, so it's not clear how the 120 and 125 are ending up in different buckets if they are written back-to-back. Need to dig into this.

--- a/agent/consul/state/tombstone_gc_test.go
+++ b/agent/consul/state/tombstone_gc_test.go
@@ -1,7 +1,6 @@
 package state
 
 import (
-	"os"
 	"testing"
 	"time"
 )
@@ -24,12 +23,8 @@ func TestTombstoneGC_invalid(t *testing.T) {
 }
 
 func TestTombstoneGC(t *testing.T) {
-	if os.Getenv("TRAVIS") == "true" {
-		t.Skip("GC test is flaky on travis-ci (see #3670)")
-	}
-
-	ttl := 20 * time.Millisecond
-	gran := 5 * time.Millisecond
+	ttl := 200 * time.Millisecond
+	gran := 50 * time.Millisecond
 	gc, err := NewTombstoneGC(ttl, gran)
 	if err != nil {
 		t.Fatalf("err: %v", err)
@@ -60,6 +55,12 @@ func TestTombstoneGC(t *testing.T) {
 	case <-time.After(ttl * 2):
 		t.Fatalf("should get expiration")
 	}
+
+	// At this point we know we are very close to a TTL bucket
+	// so we sleep a small fraction of the bucket size in order
+	// to hit the case we desire for the next section where we
+	// hint twice into the same bucket and see them coalesce.
+	time.Sleep(gran / 5)
 
 	start2 := time.Now()
 	gc.Hint(120)

